posthog/api/utils.py

```diff
@@ -319,3 +319,64 @@ def safe_clickhouse_string(s: str) -> str:
     for match in matches:
         s = s.replace(match, match.encode("unicode_escape").decode("utf8"))
     return s
+
+
+def create_event_definitions_sql(include_actions: bool, is_enterprise: bool = False) -> str:
+    # Prevent fetching deprecated `tags` field. Tags are separately fetched in TaggedItemSerializerMixin
+    ee_model = EnterpriseEventDefinition if is_enterprise else EventDefinition
+    event_definition_fields = {
+        f'"{f.column}"' for f in ee_model._meta.get_fields() if hasattr(f, "column") and f.column != "tags"  # type: ignore
+    }
+
+    if include_actions:
+        event_definition_fields.discard("id")
+        action_fields = {
+            f'"{f.column}"'  # type: ignore
+            for f in Action._meta.get_fields()
+            if hasattr(f, "column") and f.column not in ["events", "id"]  # type: ignore
+        }
+        combined_fields = event_definition_fields.union(action_fields)
+
+        raw_event_definition_fields = ",".join(
+            [f"NULL AS {col}" if col in action_fields - event_definition_fields else col for col in combined_fields]
+            + ["id", "NULL AS action_id"]
+        )
+        raw_action_fields = ",".join(
+            [f"NULL AS {col}" if col in event_definition_fields - action_fields else col for col in combined_fields]
+            + ["NULL AS id", "id AS action_id"]
+        )
+
+        event_definition_table = (
+            f"""
+                SELECT {raw_event_definition_fields}
+                    FROM ee_enterpriseeventdefinition
+                    FULL OUTER JOIN posthog_eventdefinition ON posthog_eventdefinition.id=ee_enterpriseeventdefinition.eventdefinition_ptr_id
```

Comment:
Oh, i see there's a team filter at the top level after this is returned. I think you should still add a team filter in the subqueries here for readability and depending on whether or not postgres does pushdowns intelligently. For example, the actions query below this comment looks like it selects across all teams since the `where team =` clause won't be applicable until the outer level